sdks/router-sdk/src/utils/pathCurrency.ts

```diff
@@ -0,0 +1,36 @@
+import { Currency, CurrencyAmount, Token } from '@uniswap/sdk-core'
+import { Pool as V4Pool } from '@uniswap/v4-sdk'
+import { TPool } from './TPool'
+
+export function amountWithPathCurrency(amount: CurrencyAmount<Currency>, pool: TPool): CurrencyAmount<Currency> {
+  return CurrencyAmount.fromFractionalAmount(
+    getPathCurrency(amount.currency, pool),
+    amount.numerator,
+    amount.denominator
+  )
+}
+
+export function getPathCurrency(currency: Currency, pool: TPool): Currency {
+  // return currency if the currency matches a currency of the pool
+  if (pool.involvesToken(currency as Token)) {
+    return currency
+
+    // return if currency.wrapped if pool involves wrapped currency
+  } else if (pool.involvesToken(currency.wrapped as Token)) {
+    return currency.wrapped
+
+    // return native currency if pool involves native version of wrapped currency (only applies to V4)
+  } else if (pool instanceof V4Pool) {
+    if (pool.currency0.wrapped.equals(currency)) {
+      return pool.token0
```

Comment:
i think this reads weirdly because its not apparent that `currency0` and `token0` are the same thing